src/libs/utils.js

```diff
@@ -368,3 +368,13 @@ export const truncateInteger = integer => {
 
   return `${Math.floor(integer / 1000)}k`
 }
+
+// can this be done with usePollingEffect?
+export const poll = async (pollFn, pollTime, pollUntil) => {
+  let result = await pollFn()
+  while (!pollUntil(result)) {
```

Comment:
Maybe a `do` `while` would let you avoid duplicating `await pollFn()`?

Reply:
Discussed offline - it depends on where the logic is put for whether it polls immediately. (either here or in the delay)

This follows existing code style based on `usePollingEffect` so opted to stick with present styling